controlplane/pkg/apis/connectioncontext/connectioncontext.proto

```diff
@@ -45,18 +45,36 @@ message ExtraPrefixRequest {
     uint32 requested_number = 4;
 }
 
-message ConnectionContext {
-    string src_ip_addr = 1;             /* source ip address + prefix in format <address>/<prefix> */
-    string dst_ip_addr = 2;             /* destination ip address + prefix in format <address>/<prefix> */
+message IPContext {
+    string src_ip_addr = 1; /* source ip address + prefix in format <address>/<prefix> */
+    string dst_ip_addr = 2; /* destination ip address + prefix in format <address>/<prefix> */
 
-    bool src_ip_required = 3;           /* if specified src_ip is required from NSE. */
-    bool dst_ip_required = 4;           /* if specified dst_ip is required from NSE */
+    bool src_ip_required = 3; /* if specified src_ip is required from NSE. */
+    bool dst_ip_required = 4; /* if specified dst_ip is required from NSE */
 
-    repeated Route routes = 5;          /* a list of routes via dst_ip_addr as gateway */
+    repeated Route routes = 5; /* a list of routes via dst_ip_addr as gateway */
     repeated string excluded_prefixes = 6; /* a list of excluded passed to NSE */
 
     repeated IpNeighbor ip_neighbors  = 7; /* A list of ARP table neighbours addresses */
 
-    repeated ExtraPrefixRequest extra_prefix_request = 8;   /* A request for NSE to provide extra prefixes */
+    repeated ExtraPrefixRequest extra_prefix_request = 8; /* A request for NSE to provide extra prefixes */
     repeated string extra_prefixes = 9; /* A list of extra prefixes requested */
-}
\ No newline at end of file
+    DNSConfig dns_config = 10;
+}
+
+message ConnectionContext {
+    IPContext ip_context = 1;  /* IP related context */
+    DNSConfig dns_config = 2;
+}
+
+message DNSConfig {
+    /* ips of DNS Servers for this DNSConfig.  Any given IP may be IPv4 or IPv6 */
+    repeated string dns_server_ips = 1;
+    /* domains for which this DNSConfig provides resolution.  If empty, all domains. */
+    repeated string resolves_domains = 2;
+    /* true if DNSConfig should be used to resolve listed domains listed in
+     * resolves_domains over other resolvers that did not request priority.  true is
+     * invalid if resolves_domains is empty.
+     */
+    bool prioritize = 3;
+}
```

Comment:
Can you please move the ConnectionContext at the bottom here, just for clarity. Otherwise loks great IMHO ;)

Reply:
+1 for moving ConnectionContext to the bottom :)

Reply:
done